### PR TITLE
Add scroll to bottom button

### PR DESCRIPTION
**Pull request by ma12-co**
- [x] Add the button and make it scroll down;
- [x] Fix delay in appearing/disappearing of the button;
- [x] Fix transition (fade in and out of the button);
- [x] Animate scroll;

Peek 2020-07-21 21-19

Also adds smooth scroll to the whole messageslist

Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

File: src/components/MessagesList/MessagesList.vue [hunk 1]
```diff
@@ -50,6 +50,13 @@ get the messagesList array and loop through the list to generate the messages.
 				v-for="n in 15"
 				:key="n" />
 		</template>
+		<transition name="fade">
+			<button v-show="!isScrolledToBottom"
+				class="scroll-to-bottom"
+				@click="scrollToBottom">
+				<ChevronDown :size="24" fill-color="#fff" />
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesList.vue:57]: only works when the color on primary is "white"

File: src/components/MessagesList/MessagesList.vue [hunk 2]
```diff
@@ -580,6 +590,8 @@ export default {
 </script>
 
 <style lang="scss" scoped>
+@import '../../assets/variables.scss';
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesList.vue:593]: needed?

File: src/components/MessagesList/MessagesList.vue (hunk 1 above)
Comment (ma12-co) [reply src/components/MessagesList/MessagesList.vue:57]: We're going to have a design hack Tuesday next week and by the end of that we'll have a way to use material design icons with theming

File: src/components/MessagesList/MessagesList.vue (hunk 2 above)
Comment (ma12-co) [reply src/components/MessagesList/MessagesList.vue:593]: Yes, for the fade transition

File: src/components/MessagesList/MessagesList.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/MessagesList/MessagesList.vue:57]: `var(--color-primary-text)` it is !

**Comment by nickvergessen** [comment]
Fixed the color thing and a "random" bug with the button not disappearing correctly.

File: src/components/MessagesList/MessagesList.vue [hunk 3]
```diff
@@ -51,6 +51,13 @@ get the messagesList array and loop through the list to generate the messages.
 				type="messages"
 				:count="15" />
 		</template>
+		<transition name="fade">
+			<button v-show="!isScrolledToBottom"
+				class="scroll-to-bottom"
+				@click="scrollToBottom">
+				<ChevronDown :size="24" fill-color="#fff" />
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesList.vue:58]: Tooltip says `Chevron Down icon`.
We have the same issue in quite some other places (overflowing video list is `Chevron Right icon` etc.
I guess we need to set the title ourselves

File: src/components/MessagesList/MessagesList.vue [hunk 4]
```diff
@@ -51,6 +51,13 @@ get the messagesList array and loop through the list to generate the messages.
 				type="messages"
 				:count="15" />
 		</template>
+		<transition name="fade">
+			<button v-show="!isScrolledToBottom"
+				class="scroll-to-bottom"
+				@click="scrollToBottom">
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesList.vue:57]: aria?

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**